src/slices/requests.ts

```diff
@@ -0,0 +1,60 @@
+import { createSlice, PayloadAction } from '@reduxjs/toolkit'
+import * as api from '../api/requests'
+import { AppThunk } from 'app/store'
+import { RootState } from 'app/rootReducer'
+
+interface IRequestState {
+  isLoading: boolean
+  requests: api.IRequest[]
+  error: string
+}
+
+const initialState: IRequestState = {
+  isLoading: false,
+  requests: [],
+  error: '',
+}
+
+const requestsSlice = createSlice({
```

Comment:
Notice how _createSlice_ combines _reducer_ & _actions_ into one construct. This helps reduce the need to creating multiple files, one for _action_ & one for _reducer_. The advantage of this pattern is you logically group parts that are meant to be together thus requiring less mental juggling. There is also less code involved.